### ISSUE TITLE
bug checking the ASTCENC_FLG_DECOMPRESS_ONLY flag in astcenc_context_alloc()

### ISSUE THREAD

**Issue by richgel999** [ARM-software/astc-encoder#437]
Hi, 
This looks like a simple bug in astc_entry.cpp, function `astcenc_context_alloc()`:
https://github.com/ARM-software/astc-encoder/blob/main/Source/astcenc_entry.cpp#L701

```
#if !defined(ASTCENC_DECOMPRESS_ONLY)
	// Do setup only needed by compression
	if (!(status & ASTCENC_FLG_DECOMPRESS_ONLY))
	{
```

`status` here is guaranteed to always be `ASTCENC_SUCCESS` (0) here. Pretty sure this conditional should be:

```
if (!(ctx->config.flags & ASTCENC_FLG_DECOMPRESS_ONLY))
```

**Comment by solidpixel** [comment]
Indeed it should, thanks Rich.

**Comment by solidpixel** [comment]
Fixed in 1943607ac9f73371e2115c4cd1829c24b8c8fb5c
